src/sqlfluff/core/rules/base.py

```diff
@@ -337,16 +339,29 @@ def crawl(
 
         new_lerrs = []
         new_fixes = []
+
+        def _process_lint_result(res):
+            self.discard_unsafe_fixes(res, templated_file)
+            lerr = res.to_linting_error(rule=self)
+            ignored = False
+            if lerr:
+                if ignore_mask:
+                    filtered = LintedFile.ignore_masked_violations([lerr], ignore_mask)
+                    if not filtered:
+                        lerr = None
+                        ignored = True
+            if lerr:
+                new_lerrs.append(lerr)
+            if not ignored:
+                new_fixes.extend(res.fixes)
+
         if res is None:
             # Assume this means no problems (also means no memory)
             pass
         elif isinstance(res, LintResult):
             # Extract any memory
             memory = res.memory
-            self.discard_unsafe_fixes(res, templated_file)
```

Comment:
Extracted duplicate code from here and the section below into the new `_process_lint_result()` function above, which now _also_ handles "noqa".